### PR TITLE
TextEdit: Scroll search results to the center

### LINKED ISSUE #5269: Search in script should scroll line to the center
Tested on Godot 2.0.3, Windows 10 64 bits:

When I search something in a script, the editor scrolls down to that line, but I cannot see what's next. This is not convenient because I loose half of the context. 
It would be better to scroll the line to the center of the view, like any other editor does (Sublime; Npp++, VS...).

godot_search_scroll_down

Could be referred in https://github.com/godotengine/godot/issues/2982

### PR DESCRIPTION
**Pull request by neikeq**
Closes #5269

### CODE REVIEW

File: scene/gui/text_edit.h [hunk 1]
```diff
@@ -394,8 +393,10 @@ class TextEdit : public Control  {
 	}
 	void set_auto_indent(bool p_auto_indent);
 
-	void cursor_set_column(int p_col, bool p_adjust_viewport=true);
-	void cursor_set_line(int p_row, bool p_adjust_viewport=true);
+	void adjust_viewport_to_cursor(bool p_center_view=false);
```
Comment (neikeq) [inline scene/gui/text_edit.h]: NOTE: I had to make this method public.

**Comment by neikeq** [comment]
`adjust_viewport_to_cursor(true)` is kind of confusing. I would prefer using an enum for the parameter. WDYT?

**Comment by akien-mga** [comment]
> `adjust_viewport_to_cursor(true)` is kind of confusing. I would prefer using an enum for the parameter. WDYT?

Sounds good to me. What would be the possible values? `ADJUST_TOP`, `ADJUST_CENTER` and `ADJUST_BOTTOM`?

**Comment by neikeq** [comment]
The possible values would be just `ADJUST_TOP_BOTTOM` (I am really bad with names. It's the default behaviour) and `ADJUST_CENTER`.
The idea is to avoid [Boolean traps](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap).

**Comment by neikeq** [comment]
I just ended adding a `center_viewport_to_cursor()` method. It didn't make sense to expose the other method as public.

**Pull request merged**